### PR TITLE
Allow ts-migrate to target specific files instead of the whole project

### PR DESCRIPTION
**Pull request by salemhilal**
Hey there!

I'm working on migrating pieces of Etsy's codebase to TypeScript. This project seemed like a good resource for exactly that, but we have some restrictions that prevent an all-at-once migration just yet. [I noticed](https://github.com/airbnb/ts-migrate/blob/master/packages/ts-migrate/README.md#faq) that you might be open to a contribution to enable migrating individual files, and this PR is that contribution.

### What is the change?

This PR adds a `--sources` flag that accepts a string or a glob expression representing a path to migrate. The path has to be relative to the root directory of the project. `--sources` can be used multiple times to provide multiple paths.

Using it looks like this right now:

```sh
yarn ts-migrate migrate \                  # "rename" works too
    --sources="modules/to/migrate/**/*" \  # The files we want to migrate
    --sources="node_modules/**/*.d.ts" \   # Ambient types from node_modules
    --sources="@types/**/*.d.ts" \         # Ambient types from our repo
    /path/to/our/repository
```

Specifying sources tells `ts-migrate` to ignore the sources defined in a project's `tsconfig.json` in favor of the ones passed in. The one downside of this is that ambient type files aren't loaded. `ts-migrate` works mostly ok without them, but it yells about globals that it can't identify, as you might expect. Adding ambient types back is as easy as passing another `--sources` parameter, like I did in the example above.

### What isn't done yet?

This appears to work as expected for me, but this PR still probably need these updates:
- [ ] `ts-migrate-full` doesn't know about `--sources` yet, which might be useful
- [ ] `rename` doesn't account for tsconfigs with a `files` array yet (I assume I need to just unset it, but I want to test that on a config that actually has a `files` property).
- [ ] Updating docs and writing tests, of course.

### Questions and notes.

This PR isn't fully finished, but I wanted to share the changes earlier rather than later to get some feedback on whether or not this is a good approach. Here are some notes:
- I opted to keep using the tsconfig in `rename`, rather than bypassing `findJsFiles` altogether, to ensure that whatever validation `ts.parseJsonConfigFileContent` performs continues to get applied.
- In `ts-migrate-server`, I used `addFilesFromTsConfig` when initializing the project, rather than starting the project as normal and filtering down to the provided source files. I did this because it sped up initialization and memory usage considerably in larger repos (I'm dealing with one that has around 10k files). In general, I assumed that using `--sources` implied that the runtime should be relatively quick when possible.

Let me know your thoughts, and thank you for open-sourcing this tool in the first place!

### CODE REVIEW

File: packages/ts-migrate-server/src/migrate/index.ts [hunk 1]
```diff
@@ -10,21 +10,41 @@ interface MigrateParams {
...
+  sources,
 }: MigrateParams): Promise<number> {
   let exitCode = 0;
 
   log.info(`TypeScript version: ${ts.version}`);
 
   const serverInitTimer = new PerfTimer();
 
+  if (sources !== undefined) {
+    if (Array.isArray(sources)) {
```
Comment (Rudeg) [inline packages/ts-migrate-server/src/migrate/index.ts:29]: nit: instead of multiple level `if` conditions, might be more readable to use a ternary here: 
```sources = Array.isArray(sources) ? sources.map((source) => path.join(rootDir, source)) : [sources]```

**Comment by Rudeg** [comment]
Hi Salem! 
Thank you for your contribution! 
I think it's the right approach. Everything looks good and reasonable to me.
Feel free to tag me when PR is finished. I'll review it :)

**Comment by salemhilal** [comment]
@Rudeg I think this is just about ready for review. Lemme know what you think!

File: packages/ts-migrate/bin/ts-migrate-full.sh [hunk 2]
```diff
@@ -76,17 +77,17 @@ maybe_commit -m "[ts-migrate][$folder_name] Init tsconfig.json file" -m 'Co-auth
...
 
 echo "
 [Step $((step_i++)) of ${step_count}] Fixing TypeScript errors...
 "
-$cli migrate $frontend_folder
+$cli migrate $frontend_folder $additional_args
 
 if [ "$should_remove_eslintrc" = "true" ]; then
-  rm -f $frontend_folder/.eslintrc.*
+  rm -f $frontend_folder/.eslintrc
```
Comment (Rudeg) [inline packages/ts-migrate/bin/ts-migrate-full.sh:90]: why did you change `.eslintrc.*` to `.eslintrc`? (for the context, #26)

**Comment by Rudeg** [comment]
One comment, otherwise looks good to me! 
Thanks for the contribution to `ts-migrate` 🎉

File: packages/ts-migrate/bin/ts-migrate-full.sh (hunk 2 above)
Comment (salemhilal) [reply packages/ts-migrate/bin/ts-migrate-full.sh:90]: Good question. We have an .eslintrc.js that the script wasn't detecting (my guess is that I have different glob settings configured somewhere). This meant that an additional .eslintrc would get created, and then both of them would be deleted. 

I updated the check on line 70 to use stat, which is allegedly more portable. In addition, since this script only creates .eslintrc, I updated the deletion step here to only delete that file specifically. The update to the deletion step isn't necessary, but the changes on line 70 keep our .eslintrc.js from being deleted.

**Comment by salemhilal** [comment]
My pleasure! (And congrats on the IPO!)

File: packages/ts-migrate/bin/ts-migrate-full.sh (hunk 2 above)
Comment (Rudeg) [reply packages/ts-migrate/bin/ts-migrate-full.sh:90]: great, thanks!

**Review by Rudeg — APPROVED** [review]
